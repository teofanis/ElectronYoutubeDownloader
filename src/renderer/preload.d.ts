import { IpcRequest, PromisedIpcResponse } from 'interfaces';
import { Channels } from 'main/preload';

declare global {
  interface Window {
    electron: {
      store: {
        getState: <T>() => T;
        setState: <T>(
          recipe: (draft: T) => void,
          description?: string | undefined
        ) => T;
        subscribe: <T>(
          listener: (state: T, description?: string | undefined) => void
        ) => () => void;
      };
      ipc: {
        send: <T>(
          channel: string,
          request: IpcRequest
        ) => PromisedIpcResponse<T>;
      };

      ipcRenderer: {
        removeAllListeners(channel?: Channels | string | null): void;
        sendMessage(channel: Channels, args: unknown[]): void;
        on(
          channel: string,
          func: (...args: unknown[]) => void
        ): (() => void) | undefined;
        once(channel: string, func: (...args: unknown[]) => void): void;
      };
    };
  }
}

export {};
