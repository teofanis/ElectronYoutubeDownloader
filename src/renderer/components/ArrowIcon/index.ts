// eslint-disable-next-line import/prefer-default-export
export { default as ArrowIcon } from './ArrowIcon';
