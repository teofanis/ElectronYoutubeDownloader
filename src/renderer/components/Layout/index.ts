/* eslint-disable import/prefer-default-export */
export { default as Layout } from './Layout';
