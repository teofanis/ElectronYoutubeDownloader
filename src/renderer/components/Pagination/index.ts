// eslint-disable-next-line import/prefer-default-export
export { default as Pagination } from './Pagination';
