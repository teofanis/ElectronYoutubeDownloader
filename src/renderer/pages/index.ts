import { Home } from './Home';

// eslint-disable-next-line import/prefer-default-export
export { Home };
