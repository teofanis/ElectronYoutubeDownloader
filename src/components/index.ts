// eslint-disable-next-line import/prefer-default-export
import { Button } from 'components/Button';
import { CancelButton } from 'components/CancelButton';
import { DownloadableItem } from 'components/DownloadableItem';
import { DownloadableItemControls } from 'components/DownloadableItemControls';
import { DownloadableItemTransition } from 'components/DownloadableItemTransition';
import { DownloadableItemTransitionContainer } from 'components/DownloadableItemTransitionContainer';
import { DownloadButton } from 'components/DownloadButton';
import { Downloader } from 'components/Downloader';
import { DownloadItemControl } from 'components/DownloadItemControl';
import { FileInput } from 'components/FileInput';
import { Heading } from 'components/Heading';
import { InputError } from 'components/InputError';
import { Layout } from 'components/Layout';
import { Logo } from 'components/Logo';
import { ProgressBar } from 'components/ProgressBar';
import { TextInput } from 'components/TextInput';
import { DownloadableItemTransition } from 'components/DownloadableItemTransition';
// Automatic import of components
export {
  Layout,
  Heading,
  Logo,
  Downloader,
  TextInput,
  FileInput,
  Button,
  DownloadButton,
  InputError,
  ProgressBar,
  CancelButton,
  DownloadableItem,
  DownloadableItemControls,
  DownloadItemControl,
  DownloadableItemTransitionContainer,
  DownloadableItemTransition,
  DownloadableItemTransition,
  // Automatic export of components
};
