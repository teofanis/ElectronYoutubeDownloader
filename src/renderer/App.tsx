import { Home } from 'pages';
import { MemoryRouter as Router, Route, Routes } from 'react-router-dom';
import './global.css';

export default function App() {
  return (
    <Router>
      <Routes>
        <Route path="/" element={<Home />} />
      </Routes>
    </Router>
  );
}
