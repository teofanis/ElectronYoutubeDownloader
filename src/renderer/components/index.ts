// eslint-disable-next-line import/prefer-default-export
import { Button } from 'renderer/components/Button';
import { CancelButton } from 'renderer/components/CancelButton';
import { DownloadableItem } from 'renderer/components/DownloadableItem';
import { DownloadableItemControls } from 'renderer/components/DownloadableItemControls';
import { DownloadableItemTransition } from 'renderer/components/DownloadableItemTransition';
import { DownloadableItemTransitionContainer } from 'renderer/components/DownloadableItemTransitionContainer';
import { DownloadButton } from 'renderer/components/DownloadButton';
import { Downloader } from 'renderer/components/Downloader';
import { DownloaderControls } from 'renderer/components/DownloaderControls';
import { DownloadItemControl } from 'renderer/components/DownloadItemControl';
import { FileInput } from 'renderer/components/FileInput';
import { Heading } from 'renderer/components/Heading';
import { InputError } from 'renderer/components/InputError';
import { Layout } from 'renderer/components/Layout';
import { Logo } from 'renderer/components/Logo';
import { ProgressBar } from 'renderer/components/ProgressBar';
import { TextInput } from 'renderer/components/TextInput';
import { Pagination } from 'renderer/components/Pagination';
// Automatic import of components
export {
  Layout,
  Heading,
  Logo,
  Downloader,
  TextInput,
  FileInput,
  Button,
  DownloadButton,
  InputError,
  ProgressBar,
  CancelButton,
  DownloadableItem,
  DownloadableItemControls,
  DownloadItemControl,
  DownloadableItemTransitionContainer,
  DownloadableItemTransition,
  DownloaderControls,
  Pagination,
  // Automatic export of components
};
