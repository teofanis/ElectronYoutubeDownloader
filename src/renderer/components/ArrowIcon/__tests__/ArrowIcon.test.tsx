import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { ArrowIcon } from 'renderer/components';

describe('ArrowIcon Test', () => {
  it('should render', () => {
    expect(render(<ArrowIcon type="left" />)).toBeTruthy();
  });
});
