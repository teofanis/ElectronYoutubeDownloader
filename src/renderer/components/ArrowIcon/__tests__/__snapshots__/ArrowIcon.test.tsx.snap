// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowIcon Test should render left arrow 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="w-5 h-5"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`ArrowIcon Test should render right arrow 1`] = `
<div>
  <svg
    aria-hidden="true"
    class="w-5 h-5"
    fill="currentColor"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;
